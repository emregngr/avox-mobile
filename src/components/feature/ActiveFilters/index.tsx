import { MaterialCommunityIcons } from '@expo/vector-icons'
import React, { memo, useCallback, useMemo, useRef } from 'react'
import { ScrollView, TouchableOpacity, View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import { getLocale } from '@/locales/i18next'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import { getActiveFilterLabel } from '@/utils/feature/getActiveFilterLabel'

interface ActiveFiltersProps {
  filters: any
  onClearAll: () => void
  onRemove: (key: string) => void
}

const AUTO_SCROLL_DELAY = 300

export const ActiveFilters = memo(({ filters, onClearAll, onRemove }: ActiveFiltersProps) => {
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const getActiveFilterLabelValue = getActiveFilterLabel()

  const scrollViewRef = useRef<ScrollView>(null)

  const getFilterCount = useCallback(() => {
    let count = 0
    if (!filters) return 0
    Object?.entries(filters)?.forEach(([key, value]) => {
      if (Array?.isArray(value)) {
        count += value?.length
      } else if (value !== null && value !== undefined && value !== false) {
        count += 1
      }
    })
    return count
  }, [filters])

  const filterCount = useMemo(() => getFilterCount(), [getFilterCount])

  const filterEntries = useMemo(
    () =>
      Object.entries(filters).filter(
        ([_, value]) => value !== null && value !== undefined && value !== false,
      ),
    [filters],
  )

  const handleRemove = useCallback(
    (key: string) => {
      const elementIndex = filterEntries.findIndex(([k, _]) => k === key)
      const isInLastThree = elementIndex >= filterEntries?.length - 3

      onRemove(key)

      if (isInLastThree) {
        setTimeout(() => {
          scrollViewRef?.current?.scrollToEnd({ animated: true })
        }, AUTO_SCROLL_DELAY)
      }
    },
    [filterEntries, onRemove],
  )

  if (filterCount === 0) return null

  return (
    <View className="mr-12">
      <ScrollView
        contentContainerClassName="items-center"
        ref={scrollViewRef}
        showsHorizontalScrollIndicator={false}
        horizontal
      >
        <View className="flex-row items-center pr-4">
          <TouchableOpacity
            activeOpacity={0.7}
            className="bg-error px-3 py-1.5 rounded-full overflow-hidden mr-8 left-6"
            hitSlop={10}
            onPress={onClearAll}
            testID="active-filters-clear-button"
          >
            <ThemedText color="text-100" type="body3">
              {getLocale('clear')} ({filterCount})
            </ThemedText>
          </TouchableOpacity>

          {filterEntries.map(([key, filterValue]) => (
            <View
              className="bg-secondary-100 px-3 py-1.5 rounded-full overflow-hidden mr-2 flex-row items-center"
              key={key}
            >
              <ThemedText color="text-100" type="body3">
                {getActiveFilterLabelValue(key, filterValue)}
              </ThemedText>
              <TouchableOpacity
                activeOpacity={0.7}
                className="bg-background-quaternary rounded-full overflow-hidden ml-2"
                hitSlop={10}
                onPress={() => handleRemove(key)}
                testID={`active-filters-remove-${key}`}
              >
                <MaterialCommunityIcons color={colors?.onPrimary100} name="close" size={14} />
              </TouchableOpacity>
            </View>
          ))}
        </View>
      </ScrollView>
    </View>
  )
})
