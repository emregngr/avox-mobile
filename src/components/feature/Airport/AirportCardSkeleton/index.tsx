import React, { memo, useEffect } from 'react'
import { View } from 'react-native'
import Animated, {
  interpolate,
  useAnimatedStyle,
  useSharedValue,
  withRepeat,
  withTiming,
} from 'react-native-reanimated'

const STATIC_STYLES = {
  badge: {
    height: 40,
    left: 4,
    position: 'absolute' as const,
    top: 4,
    width: 40,
  },
  image: {
    height: 130,
  },
}

export const AirportCardSkeleton = memo(() => {
  const pulse = useSharedValue<number>(0)

  useEffect(() => {
    pulse.value = withRepeat(withTiming(1, { duration: 1000 }), -1, true)
  }, [pulse])

  const animatedStyle = useAnimatedStyle(() => ({
    opacity: interpolate(pulse.value, [0, 1], [0.3, 0.7]),
  }))

  return (
    <View
      className="bg-background-secondary rounded-xl mb-4 w-[48%] border border-background-quaternary shadow shadow-background-quaternary"
      testID="skeleton-block"
    >
      <View className="rounded-t-xl overflow-hidden w-full justify-center">
        <Animated.View
          className="w-full bg-background-primary"
          style={[STATIC_STYLES.image, animatedStyle]}
        />

        <Animated.View
          className="bg-background-quaternary rounded overflow-hidden"
          style={[STATIC_STYLES.badge, animatedStyle]}
        />

        <Animated.View
          className="absolute bottom-2 left-2 w-12 h-6 bg-background-quaternary rounded-xl overflow-hidden"
          style={animatedStyle}
        />

        <Animated.View
          className="absolute bottom-2 right-2 w-14 h-6 bg-background-quaternary rounded-xl overflow-hidden"
          style={animatedStyle}
        />

        <Animated.View
          className="absolute top-2 right-2 w-8 h-8 bg-background-quaternary rounded-full overflow-hidden"
          style={animatedStyle}
        />
      </View>

      <View className="px-3 py-3">
        <View className="h-56 justify-between">
          <View>
            <Animated.View
              className="h-6 bg-background-quaternary rounded-xl overflow-hidden"
              style={animatedStyle}
            />

            <Animated.View
              className="h-4 bg-background-quaternary rounded-xl overflow-hidden w-3/4 mt-2"
              style={animatedStyle}
            />
          </View>

          <View className="bg-background-secondary rounded-xl p-2 shadow shadow-background-quaternary mt-2 border border-background-quaternary">
            <View className="flex-row items-center justify-between">
              <View className="flex-1 items-center">
                <Animated.View
                  className="h-6 w-8 bg-background-quaternary rounded overflow-hidden mb-1"
                  style={animatedStyle}
                />
                <Animated.View
                  className="h-3 w-10 bg-background-quaternary rounded overflow-hidden"
                  style={animatedStyle}
                />
              </View>

              <View className="w-[1px] h-6 bg-background-quaternary" />

              <View className="flex-1 items-center">
                <Animated.View
                  className="h-6 w-8 bg-background-quaternary rounded overflow-hidden mb-1"
                  style={animatedStyle}
                />
                <Animated.View
                  className="h-3 w-12 bg-background-quaternary rounded overflow-hidden"
                  style={animatedStyle}
                />
              </View>
            </View>

            <View className="my-3 w-[100px] self-center flex-row justify-between">
              <View className="h-[1px] w-8 bg-background-quaternary" />
              <View className="h-[1px] w-8 bg-background-quaternary" />
            </View>

            <View className="flex-row items-center justify-between">
              <View className="flex-1 items-center">
                <Animated.View
                  className="h-6 w-10 bg-background-quaternary rounded overflow-hidden mb-1"
                  style={animatedStyle}
                />
                <Animated.View
                  className="h-3 w-14 bg-background-quaternary rounded overflow-hidden"
                  style={animatedStyle}
                />
              </View>

              <View className="w-[1px] h-6 bg-background-quaternary" />

              <View className="flex-1 items-center">
                <Animated.View
                  className="h-6 w-6 bg-background-quaternary rounded overflow-hidden mb-1"
                  style={animatedStyle}
                />
                <Animated.View
                  className="h-3 w-10 bg-background-quaternary rounded overflow-hidden"
                  style={animatedStyle}
                />
              </View>
            </View>
          </View>
        </View>
      </View>

      <View className="h-20 px-3 mb-3">
        <Animated.View
          className="h-4 w-28 bg-background-quaternary rounded overflow-hidden mb-2"
          style={animatedStyle}
        />

        <Animated.View
          className="bg-background-quaternary rounded-xl overflow-hidden mt-1 h-10"
          style={animatedStyle}
        />
      </View>
    </View>
  )
})
