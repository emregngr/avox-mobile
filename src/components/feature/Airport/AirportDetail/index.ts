export * from './AirportHeader'
export * from './AirportRowItem'
export * from './AirportSectionRow'
export * from './Tab'
