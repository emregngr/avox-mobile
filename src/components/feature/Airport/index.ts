export * from './AirportCard'
export * from './AirportCardSkeleton'
export * from './AirportDetail'
export * from './AirportsTab'
