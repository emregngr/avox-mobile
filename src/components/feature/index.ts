export * from './ActiveFilters'
export * from './Airline'
export * from './Airport'
export * from './AppNavigator'
export * from './FavoriteButton'
export * from './Favorites'
export * from './FilterModal'
export * from './Home'
