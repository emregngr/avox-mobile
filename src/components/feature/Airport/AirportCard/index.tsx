import { Image } from 'expo-image'
import { router } from 'expo-router'
import React, { memo, useCallback, useMemo } from 'react'
import { TouchableOpacity, View } from 'react-native'

import { FullScreenLoading } from '@/components/common/FullScreenLoading'
import { ThemedText } from '@/components/common/ThemedText'
import { FavoriteButton } from '@/components/feature/FavoriteButton'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airport } from '@/types/feature/airport'
import { responsive } from '@/utils/common/responsive'
import getAirportImage from '@/utils/feature/getAirportImage'
import type { AirportType } from '@/utils/feature/getBadge'
import { getAirportBadge } from '@/utils/feature/getBadge'

interface AirportCardProps {
  airport: Airport
}

const AirportCard = memo(({ airport }: AirportCardProps) => {
  const { selectedLocale } = useLocaleStore()

  const {
    facilities: { googleMapsRating },
    flightOperations: { airlines = [], destinationCount, destinationCountries },
    iataCode,
    icaoCode,
    id,
    infrastructure: { passengerCapacity },
    name,
    operations: {
      airportType,
      country,
      location: { city },
      region,
    },
  } = airport || {}

  const image = useMemo(() => getAirportImage(airportType as AirportType), [airportType])
  const badge = useMemo(() => getAirportBadge(airportType as AirportType), [airportType])

  const locationString = useMemo(() => `${city}, ${country}, ${region}`, [city, country, region])

  const airlinesString = useMemo(() => airlines?.map(airline => airline).join(', '), [airlines])

  const handlePress = useCallback(() => {
    router.navigate({
      params: {
        airport: JSON.stringify(airport),
      },
      pathname: '/airport-detail',
    })
  }, [airport])

  const localeStrings = useMemo(
    () => ({
      country: getLocale('country'),
      passenger: getLocale('passenger'),
      popularAirlines: getLocale('popularAirlines'),
      rating: getLocale('rating'),
      route: getLocale('route'),
    }),
    [selectedLocale],
  )

  const containerWidth = useMemo(() => responsive.deviceWidth / 2 - 44, [])

  return (
    <TouchableOpacity
      activeOpacity={0.7}
      className="bg-background-secondary rounded-xl mb-4 w-[48%] border border-background-quaternary shadow shadow-background-quaternary"
      hitSlop={20}
      onPress={handlePress}
    >
      <View className="rounded-t-xl w-full justify-center overflow-hidden">
        <Image
          style={{
            height: 130,
            width: '100%',
          }}
          cachePolicy="memory-disk"
          contentFit="cover"
          source={image}
          transition={0}
        />

        <Image
          style={{
            height: 40,
            left: 4,
            position: 'absolute',
            top: 4,
            width: 40,
          }}
          cachePolicy="memory-disk"
          contentFit="contain"
          source={badge}
          transition={0}
        />

        <View className="bg-primary-100 px-2 py-1 rounded-xl overflow-hidden absolute bottom-2 left-2">
          <ThemedText color="text-100" type="button2">
            {iataCode}
          </ThemedText>
        </View>

        <View className="bg-primary-100 px-2 py-1 rounded-xl overflow-hidden absolute bottom-2 right-2">
          <ThemedText color="text-100" type="button2">
            {icaoCode}
          </ThemedText>
        </View>

        <FavoriteButton id={String(id)} type="airport" />
      </View>

      <View className="px-3 py-3">
        <View className="h-56 justify-between">
          <ThemedText
            color="text-100" ellipsizeMode="tail" numberOfLines={2}
            type="h4"
          >
            {name}
          </ThemedText>

          <ThemedText
            className="mt-1"
            color="text-90"
            ellipsizeMode="tail"
            numberOfLines={2}
            type="body2"
          >
            {locationString}
          </ThemedText>

          <View
            className={
              'bg-background-secondary rounded-xl p-2 shadow shadow-background-quaternary mt-2 border border-background-quaternary'
            }
          >
            <View className="flex-row items-center justify-between">
              <View className="flex-1 items-center">
                <ThemedText className="mb-1" color="text-100" type="h4">
                  {destinationCount}
                </ThemedText>
                <ThemedText
                  color="text-90" lineBreakMode="tail" numberOfLines={1}
                  type="body4"
                >
                  {localeStrings.route}
                </ThemedText>
              </View>

              <View className="w-[1px] h-6 bg-primary-100" />

              <View className="flex-1 items-center">
                <ThemedText className="mb-1" color="text-100" type="h4">
                  {destinationCountries}
                </ThemedText>
                <ThemedText
                  color="text-90" lineBreakMode="tail" numberOfLines={1}
                  type="body4"
                >
                  {localeStrings.country}
                </ThemedText>
              </View>
            </View>

            <View className="my-3 w-[100px] self-center flex flex-row justify-between">
              <View className="h-[1px] w-8 bg-primary-100" />
              <View className="h-[1px] w-8 bg-primary-100" />
            </View>

            <View className="flex-row items-center justify-between">
              <View className="flex-1 items-center">
                <ThemedText className="mb-1" color="text-100" type="h4">
                  {passengerCapacity} m
                </ThemedText>
                <ThemedText
                  color="text-90" lineBreakMode="tail" numberOfLines={1}
                  type="body4"
                >
                  {localeStrings.passenger}
                </ThemedText>
              </View>

              <View className="w-[1px] h-6 bg-primary-100" />
              <View className="flex-1 items-center">
                <ThemedText className="mb-1" color="text-100" type="h4">
                  {googleMapsRating}
                </ThemedText>
                <ThemedText
                  color="text-90" lineBreakMode="tail" numberOfLines={1}
                  type="body4"
                >
                  {localeStrings.rating}
                </ThemedText>
              </View>
            </View>
          </View>
        </View>
      </View>

      <View className="h-20 px-3 mb-3">
        <ThemedText color="text-100" type="body3">
          {localeStrings.popularAirlines}
        </ThemedText>

        <View
          className="bg-background-quaternary px-2 py-1 rounded-xl overflow-hidden mt-2"
          style={{ width: containerWidth }}
        >
          <View className="flex-row items-center px-1">
            <ThemedText
              color="text-90" ellipsizeMode="tail" numberOfLines={2}
              type="body4"
            >
              {airlinesString}
            </ThemedText>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  )
})

AirportCard.displayName = 'AirportCard'

interface AirportsLoadMoreFooterProps {
  airportsHasNext: boolean
  airportsLoading: boolean
  flatAirportsData: Airport[]
}

const AirportsLoadMoreFooter = memo(
  ({ airportsHasNext, airportsLoading, flatAirportsData }: AirportsLoadMoreFooterProps) => {
    const { selectedLocale } = useLocaleStore()

    const localeStrings = useMemo(
      () => ({
        allAirportsShown: getLocale('allAirportsShown'),
        noAirports: getLocale('noAirports'),
      }),
      [selectedLocale],
    )

    const isInitialLoading = useMemo(
      () => airportsLoading && flatAirportsData?.length === 0,
      [airportsLoading, flatAirportsData?.length],
    )

    const isEmptyWithNoMore = useMemo(
      () => !airportsHasNext && flatAirportsData?.length === 0,
      [airportsHasNext, flatAirportsData?.length],
    )

    const isEndOfList = useMemo(
      () => !airportsHasNext && flatAirportsData?.length > 0,
      [airportsHasNext, flatAirportsData?.length],
    )

    if (isInitialLoading) {
      return (
        <View className="py-8 items-center">
          <FullScreenLoading />
        </View>
      )
    }

    if (isEmptyWithNoMore) {
      return (
        <View className="p-4 items-center">
          <View className="px-4 py-2 rounded-full overflow-hidden bg-background-tertiary items-center justify-center">
            <ThemedText color="text-100" type="body1">
              {localeStrings.noAirports}
            </ThemedText>
          </View>
        </View>
      )
    }

    if (isEndOfList) {
      return (
        <View className="p-4 items-center">
          <View className="px-4 py-2 rounded-full overflow-hidden bg-background-tertiary items-center justify-center">
            <ThemedText color="text-100" type="body1">
              {localeStrings.allAirportsShown} ({flatAirportsData?.length})
            </ThemedText>
          </View>
        </View>
      )
    }

    return null
  },
)

AirportsLoadMoreFooter.displayName = 'AirportsLoadMoreFooter'

export { AirportCard, AirportsLoadMoreFooter }
