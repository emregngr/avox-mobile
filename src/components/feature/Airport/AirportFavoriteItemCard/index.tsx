import { Image } from 'expo-image'
import { router } from 'expo-router'
import React, { memo, useCallback, useMemo } from 'react'
import { TouchableOpacity, View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import { FavoriteButton } from '@/components/feature/FavoriteButton'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airport } from '@/types/feature/airport'
import { responsive } from '@/utils/common/responsive'
import getAirportImage from '@/utils/feature/getAirportImage'
import type { AirportType } from '@/utils/feature/getBadge'
import { getAirportBadge } from '@/utils/feature/getBadge'

interface AirportFavoriteItemCardProps {
  airport: Airport
}

export const AirportFavoriteItemCard = memo(({ airport }: AirportFavoriteItemCardProps) => {
  const { selectedLocale } = useLocaleStore()

  const {
    facilities: { googleMapsRating },
    flightOperations: { airlines = [], destinationCount, destinationCountries },
    iataCode,
    icaoCode,
    id,
    infrastructure: { passengerCapacity },
    name,
    operations: {
      airportType,
      country,
      location: { city },
      region,
    },
  } = airport || {}

  const image = useMemo(() => getAirportImage(airportType as AirportType), [airportType])
  const badge = useMemo(() => getAirportBadge(airportType as AirportType), [airportType])

  const handlePress = useCallback(() => {
    router.navigate({
      params: {
        airport: JSON.stringify(airport),
      },
      pathname: '/airport-detail',
    })
  }, [airport])

  const localeStrings = useMemo(
    () => ({
      country: getLocale('country'),
      passenger: getLocale('passenger'),
      popularAirlines: getLocale('popularAirlines'),
      rating: getLocale('rating'),
      route: getLocale('route'),
    }),
    [selectedLocale],
  )

  const locationText = useMemo(() => `${city}, ${country}, ${region}`, [city, country, region])

  const airlinesText = useMemo(() => airlines?.map(airline => airline).join(', '), [airlines])

  const airlinesContainerWidth = useMemo(() => responsive.deviceWidth / 2 - 44, [])

  return (
    <TouchableOpacity
      activeOpacity={0.7}
      className="bg-background-secondary rounded-xl mb-4 w-[48%] border border-background-quaternary shadow shadow-background-quaternary"
      hitSlop={20}
      onPress={handlePress}
    >
      <View className="rounded-t-xl overflow-hidden w-full justify-center">
        <Image
          style={{
            height: 130,
            width: '100%',
          }}
          cachePolicy="memory-disk"
          contentFit="cover"
          source={image}
          transition={0}
        />

        <Image
          style={{
            height: 40,
            left: 4,
            position: 'absolute',
            top: 4,
            width: 40,
          }}
          cachePolicy="memory-disk"
          contentFit="contain"
          source={badge}
          transition={0}
        />

        <View className="bg-primary-100 px-2 py-1 rounded-xl overflow-hidden absolute bottom-2 left-2">
          <ThemedText color="text-100" type="button2">
            {iataCode}
          </ThemedText>
        </View>

        <View className="bg-primary-100 px-2 py-1 rounded-xl overflow-hidden absolute bottom-2 right-2">
          <ThemedText color="text-100" type="button2">
            {icaoCode}
          </ThemedText>
        </View>

        <FavoriteButton id={String(id)} type="airport" />
      </View>

      <View className="px-3 py-3">
        <View className="h-56 justify-between">
          <ThemedText
            color="text-100" ellipsizeMode="tail" numberOfLines={2}
            type="h4"
          >
            {name}
          </ThemedText>

          <ThemedText
            className="mt-1"
            color="text-90"
            ellipsizeMode="tail"
            numberOfLines={2}
            type="body2"
          >
            {locationText}
          </ThemedText>

          <View
            className={
              'bg-background-secondary rounded-xl p-2 shadow shadow-background-quaternary mt-2 border border-background-quaternary'
            }
          >
            <View className="flex-row items-center justify-between">
              <View className="flex-1 items-center">
                <ThemedText className="mb-1" color="text-100" type="h4">
                  {destinationCount}
                </ThemedText>
                <ThemedText
                  color="text-90" lineBreakMode="tail" numberOfLines={1}
                  type="body4"
                >
                  {localeStrings.route}
                </ThemedText>
              </View>

              <View className="w-[1px] h-6 bg-primary-100" />

              <View className="flex-1 items-center">
                <ThemedText className="mb-1" color="text-100" type="h4">
                  {destinationCountries}
                </ThemedText>
                <ThemedText
                  color="text-90" lineBreakMode="tail" numberOfLines={1}
                  type="body4"
                >
                  {localeStrings.country}
                </ThemedText>
              </View>
            </View>

            <View className="my-3 w-[100px] self-center flex flex-row justify-between">
              <View className="h-[1px] w-8 bg-primary-100" />
              <View className="h-[1px] w-8 bg-primary-100" />
            </View>

            <View className="flex-row items-center justify-between">
              <View className="flex-1 items-center">
                <ThemedText className="mb-1" color="text-100" type="h4">
                  {passengerCapacity} m
                </ThemedText>
                <ThemedText
                  color="text-90" lineBreakMode="tail" numberOfLines={1}
                  type="body4"
                >
                  {localeStrings.passenger}
                </ThemedText>
              </View>

              <View className="w-[1px] h-6 bg-primary-100" />
              <View className="flex-1 items-center">
                <ThemedText className="mb-1" color="text-100" type="h4">
                  {googleMapsRating}
                </ThemedText>
                <ThemedText
                  color="text-90" lineBreakMode="tail" numberOfLines={1}
                  type="body4"
                >
                  {localeStrings.rating}
                </ThemedText>
              </View>
            </View>
          </View>
        </View>
      </View>

      <View className="h-20 px-3 mb-3">
        <ThemedText color="text-100" type="body3">
          {localeStrings.popularAirlines}
        </ThemedText>

        <View
          className="bg-background-quaternary px-2 py-1 rounded-xl overflow-hidden mt-2"
          style={{ width: airlinesContainerWidth }}
        >
          <View className="flex-row items-center px-1">
            <ThemedText
              color="text-90" ellipsizeMode="tail" numberOfLines={2}
              type="body4"
            >
              {airlinesText}
            </ThemedText>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  )
})
