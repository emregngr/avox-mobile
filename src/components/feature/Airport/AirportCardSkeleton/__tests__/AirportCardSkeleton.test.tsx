import { render } from '@testing-library/react-native'
import React from 'react'
import * as Reanimated from 'react-native-reanimated'

import { AirportCardSkeleton } from '@/components/feature/Airport'

jest.mock('react-native-reanimated', () => {
  const { View } = require('react-native')
  const Reanimated = require('react-native-reanimated/mock')

  return {
    ...Reanimated,
    useSharedValue: jest.fn(),
    useAnimatedStyle: jest.fn(),
    withTiming: jest.fn(),
    withRepeat: jest.fn(),
    Animated: {
      ...Reanimated.Animated,
      View,
    },
  }
})

const mockedReanimatedUseSharedValue = Reanimated.useSharedValue as jest.Mock
const mockedReanimatedUseAnimatedStyle = Reanimated.useAnimatedStyle as jest.Mock
const mockedReanimatedWithTiming = Reanimated.withTiming as jest.Mock
const mockedReanimatedWithRepeat = Reanimated.withRepeat as jest.Mock

beforeEach(() => {
  mockedReanimatedUseSharedValue.mockReturnValue({ value: 0.5 })
  mockedReanimatedUseAnimatedStyle.mockImplementation(styleFactory => styleFactory())
  mockedReanimatedWithTiming.mockImplementation(value => value)
  mockedReanimatedWithRepeat.mockImplementation(animation => animation)
})

describe('AirportCard Component', () => {
  it('calls withRepeat + withTiming to start pulse animation', () => {
    render(<AirportCardSkeleton />)

    expect(Reanimated.withTiming).toHaveBeenCalledWith(1, { duration: 1000 })
    expect(Reanimated.withRepeat).toHaveBeenCalledWith(1, -1, true)
  })

  it('renders multiple skeleton blocks', () => {
    const { getAllByTestId } = render(<AirportCardSkeleton />)
    expect(getAllByTestId('skeleton-block').length).toBeGreaterThan(0)
  })
})

describe('AirportCardSkeleton Component Snapshot', () => {
  it('should render the AirportCardSkeleton Component successfully', () => {
    const { toJSON } = render(<AirportCardSkeleton />)

    expect(toJSON()).toMatchSnapshot()
  })
})
