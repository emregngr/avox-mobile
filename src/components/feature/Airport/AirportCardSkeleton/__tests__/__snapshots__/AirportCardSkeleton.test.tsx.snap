// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AirportCardSkeleton Component Snapshot should render the AirportCardSkeleton Component successfully 1`] = `
<View
  className="bg-background-secondary rounded-xl mb-4 w-[48%] border border-background-quaternary shadow shadow-background-quaternary"
  testID="skeleton-block"
>
  <View
    className="rounded-t-xl overflow-hidden w-full justify-center"
  >
    <View
      className="w-full bg-background-primary"
      style={
        [
          {
            "height": 130,
          },
          {
            "opacity": undefined,
          },
        ]
      }
    />
    <View
      className="bg-background-quaternary rounded overflow-hidden"
      style={
        [
          {
            "height": 40,
            "left": 4,
            "position": "absolute",
            "top": 4,
            "width": 40,
          },
          {
            "opacity": undefined,
          },
        ]
      }
    />
    <View
      className="absolute bottom-2 left-2 w-12 h-6 bg-background-quaternary rounded-xl overflow-hidden"
      style={
        {
          "opacity": undefined,
        }
      }
    />
    <View
      className="absolute bottom-2 right-2 w-14 h-6 bg-background-quaternary rounded-xl overflow-hidden"
      style={
        {
          "opacity": undefined,
        }
      }
    />
    <View
      className="absolute top-2 right-2 w-8 h-8 bg-background-quaternary rounded-full overflow-hidden"
      style={
        {
          "opacity": undefined,
        }
      }
    />
  </View>
  <View
    className="px-3 py-3"
  >
    <View
      className="h-56 justify-between"
    >
      <View>
        <View
          className="h-6 bg-background-quaternary rounded-xl overflow-hidden"
          style={
            {
              "opacity": undefined,
            }
          }
        />
        <View
          className="h-4 bg-background-quaternary rounded-xl overflow-hidden w-3/4 mt-2"
          style={
            {
              "opacity": undefined,
            }
          }
        />
      </View>
      <View
        className="bg-background-secondary rounded-xl p-2 shadow shadow-background-quaternary mt-2 border border-background-quaternary"
      >
        <View
          className="flex-row items-center justify-between"
        >
          <View
            className="flex-1 items-center"
          >
            <View
              className="h-6 w-8 bg-background-quaternary rounded overflow-hidden mb-1"
              style={
                {
                  "opacity": undefined,
                }
              }
            />
            <View
              className="h-3 w-10 bg-background-quaternary rounded overflow-hidden"
              style={
                {
                  "opacity": undefined,
                }
              }
            />
          </View>
          <View
            className="w-[1px] h-6 bg-background-quaternary"
          />
          <View
            className="flex-1 items-center"
          >
            <View
              className="h-6 w-8 bg-background-quaternary rounded overflow-hidden mb-1"
              style={
                {
                  "opacity": undefined,
                }
              }
            />
            <View
              className="h-3 w-12 bg-background-quaternary rounded overflow-hidden"
              style={
                {
                  "opacity": undefined,
                }
              }
            />
          </View>
        </View>
        <View
          className="my-3 w-[100px] self-center flex-row justify-between"
        >
          <View
            className="h-[1px] w-8 bg-background-quaternary"
          />
          <View
            className="h-[1px] w-8 bg-background-quaternary"
          />
        </View>
        <View
          className="flex-row items-center justify-between"
        >
          <View
            className="flex-1 items-center"
          >
            <View
              className="h-6 w-10 bg-background-quaternary rounded overflow-hidden mb-1"
              style={
                {
                  "opacity": undefined,
                }
              }
            />
            <View
              className="h-3 w-14 bg-background-quaternary rounded overflow-hidden"
              style={
                {
                  "opacity": undefined,
                }
              }
            />
          </View>
          <View
            className="w-[1px] h-6 bg-background-quaternary"
          />
          <View
            className="flex-1 items-center"
          >
            <View
              className="h-6 w-6 bg-background-quaternary rounded overflow-hidden mb-1"
              style={
                {
                  "opacity": undefined,
                }
              }
            />
            <View
              className="h-3 w-10 bg-background-quaternary rounded overflow-hidden"
              style={
                {
                  "opacity": undefined,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    className="h-20 px-3 mb-3"
  >
    <View
      className="h-4 w-28 bg-background-quaternary rounded overflow-hidden mb-2"
      style={
        {
          "opacity": undefined,
        }
      }
    />
    <View
      className="bg-background-quaternary rounded-xl overflow-hidden mt-1 h-10"
      style={
        {
          "opacity": undefined,
        }
      }
    />
  </View>
</View>
`;
